core/src/main/java/bisq/core/dao/state/DaoStateService.java

```diff
@@ -115,6 +116,9 @@ public void applySnapshot(DaoState snapshot) {
 
         daoState.setChainHeight(snapshot.getChainHeight());
 
+        daoState.getTxMap().clear();
+        daoState.getTxMap().putAll(snapshot.getTxMap());
```

Comment:
How did caching at the `DaoState` level compare to caching at the `Block` level? Keeping object in-sync is complicated and I'd be interested in understanding if the simpler block-level cache has most of the gain without any of the synchronization complication.

Reply:
I'm not quite sure what you mean by caching at the block level - do you mean adding a transient field of some kind to Block?

Reply:
Yes. Instead of keeping a map of all txns that needs to be kept in sync in the `DaoState` object, the `Block` could just cache the list of transactions for itself and the lookup functions changed from O(txns) to O(blocks). I only bring this up because it would require less complexity and since there are no tests it may make it easier to guarantee correctness through just code review.

Reply:
I fear that might not have much effect on performance as usually there are very few txs in a block. In average we have 1 tx in 2 blocks.